# the analysis report

(To run this Jupyter `.ipynb` file again, go [here](https://github.com/jupyterlab/jupyterlab#jupyterlab), click on the '`launch binder`' badge and the when the session comes up, upload this notebook.   
Or use [JupyterLite](https://github.com/jupyterlite/jupyterlite#jupyterlite) for running/developing as everything used so far runs there, with the exception of the recommended printing method.)
For the recommended pringting method. Once you have your notebooks worked out, use a session launched from [my respect_collapsed_folded_code4HTML_reports-binder repo](https://github.com/fomightez/respect_collapsed_folded_code4HTML_reports-binder) to generate the actual HTML. 

The idea being you can see how it works and plug in your own notebook content here. The script is portable actually so you can also move it to your own Jupyter/Python session.

<details>
  <summary>✚ <b><i>For printing</i></b> this notebook (Click to see details)</summary>
    <p>Printing nicely formatted showing equations & respecting collapsed code cells:</p>
    This uses an equation that won't print using the 'Print' directly.<br>
It also uses collapsed/folded cells that won't print respecting that style when using the 'Print' directly.<br>  
So I used the advice <a href="https://discourse.jupyter.org/t/equations-rendered-in-notebook-print-preview-not-in-jupyter-lab-print-preview/6081/2?u=fomightez">here</a> to use `nbconvert`. And because I didn't want the code showing, I originally used <code>--no-input</code>, i.e. <code>jupyter nbconvert --to html --no-input the_analysis.ipynb</code>. However, I only wanted the collapsed/folded cells not showing, and so I developed the script <code>convert_to_respect_collapsed_folded_code_and_allow_equations.py</code>.<br>

<p>If you are doing this in the session made for the demonstration, the converstion script will already be resent. Otherwise, put that script <code>convert_to_respect_collapsed_folded_code_and_allow_equations.py</code>  in this session, too, and then run this in the terminal:</p>

<pre><code>python convert_to_respect_collapsed_folded_code_and_allow_equations.py the_analysis.ipynb
</code></pre>

<p>Then download that HTML, open in browser, and print from local computer.<br>  
Unfortunately, `jupyter nbconvert --to html` command alone seems to move the output over to the left some, and as a result, this approach does that as well. However, it is a minor nuisance
</details>


----

-----

Fancy content to represent a report

In [1]:
# define some variables
cost_case_item = 228.36
total_items_per_case = 40


For the area of the circle, we can calculate this from the equation:

$$A = \frac{1}{4}\pi d^2$$

The description for the plates says the diameter is 150 mm.  
So the result would be:

In [2]:
#calculate the area with sympy
import sympy as sp
# Define diameter
d = 150  # mm

# Calculate area using the diameter formula
A = sp.Rational(1, 4) * sp.pi * d**2
A_cm2 = A / 100

#print(f"Area = {A}")  # Exact symbolic form
print(f"Area ≈ {float(A):.2f} mm²")  # Numerical approximation
print(" -or- ")
print(f"Area ≈ {float(A_cm2):.2f} cm²")  # Numerical approximation in centimeters

Area ≈ 17671.46 mm²
 -or- 
Area ≈ 176.71 cm²


**Area is pretty much same for some other item because almost same as 176.71 cm<sup>2</sup>.   

### Cost per item 


In [3]:
cost_per_item = cost_case_item/ttotal_items_per_case
print(f'cost per item: ${cost_per_item:1.2f}')

cost per flask: $3.22
cost per plate: $2.38


### Relatively cool

$$E = (mc)^2$$

In [ ]:
import sympy as sp

# Define symbolic variables
m0, v, c = sp.symbols('m_0 v c', positive=True, real=True)

# Define the Lorentz factor gamma
gamma = 1 / sp.sqrt(1 - (v**2 / c**2))

# Total relativistic energy
E_total = gamma * m0 * c**2

# Compute Taylor expansion up to order v^4 around v = 0
E_expanded = sp.series(E_total, v, 0, n=5).removeO()

print("Expanded Energy Expression:")
sp.pprint(E_expanded)

In [ ]:
import sympy as sp

m0, v, c, E = sp.symbols('m_0 v c E', positive=True, real=True)

# Define relativistic energy equation
gamma = 1 / sp.sqrt(1 - (v**2 / c**2))
eq = sp.Eq(E, gamma * m0 * c**2)

# Solve for velocity (v)
v_sol = sp.solve(eq, v)

print("Velocity as a function of Total Energy (E):")
sp.pprint(v_sol[0])

In [1]:
import sympy as sp

m, c = sp.symbols('m c')
E = m * c**2

# Substitute 1 gram (0.001 kg) and light speed (299,792,458 m/s)
energy_in_joules = E.subs({
    m: 0.001,                 # 1 gram in kg
    c: 299792458              # speed of light in m/s
})

print(f"Energy released from 1g of mass: {float(energy_in_joules):.4e} Joules")

ModuleNotFoundError: No module named 'sympy'

In [ ]:
import sympy as sp

# Define symbols
m, c = sp.symbols("m c")

# Define the symbolic equation e = (mc)^2
E = (m * c) ** 2

print("Symbolic Expression:", E)
# Output: m**2*c**2